src/components/SearchModal/SearchModal.js

```diff
@@ -81,23 +88,50 @@ const SearchModal = ({
   resources,
   isMultiSelect,
 }) => {
-  const [lastFetchedPage, setLastFetchedPage] = useState(0);
-  const [selectedItems, setSelectedItems] = useState([]);
-
-  const getInitialFiltersState = () => packageFilters.reduce((filtersState, filter) => ({
+  const getInitialFiltersState = searchType => searchFiltersConfig[searchType].reduce((filtersState, filter) => ({
     ...filtersState,
     [filter.name]: filter.defaultValue,
   }), {});
 
-  const hasLoaded = !!resources.packages?.hasLoaded;
-  const fetchIsPending = !hasLoaded && resources.packages?.isPending;
+  const getInitialSearchConfig = searchType => {
+    const searchConfig = {
+      selectedItems: [],
+      lastFetchedPage: 0,
+      searchQuery: '',
+      searchFilters: getInitialFiltersState(searchType),
+      searchByTagsEnabled: false,
+      searchAccessTypesEnabled: false,
+    };
+
+    if (searchType === searchTypes.TITLE) {
+      searchConfig.searchField = titleSearchFields.TITLE;
+    }
+
+    return searchConfig;
+  };
 
-  const initialFiltersState = useMemo(getInitialFiltersState, []);
-  const [searchQuery, setSearchQuery] = useState('');
-  const [searchFilters, setSearchFilters] = useState(initialFiltersState);
+  const [searchType, setSearchType] = useState(searchTypes.PACKAGE);
+  const [packageSearchConfig, setPackageSearchConfig] = useState(() => getInitialSearchConfig(searchTypes.PACKAGE));
+  const [titlesSearchConfig, setTitleSearchConfig] = useState(() => getInitialSearchConfig(searchTypes.TITLE));
 
-  const [searchByTagsEnabled, setSearchByTagsEnabled] = useState(false);
-  const [searchAccessTypesEnabled, setSearchAccessTypesEnabled] = useState(false);
+  const changeCurrentSearchConfig = updater => {
+    if (searchType === searchTypes.PACKAGE) {
```

Comment:
I think, it will be better to create two variables
`const isPackage = searchType === searchTypes.PACKAGE;
const isTitle = searchType === searchTypes.TITLE;
`